### PR TITLE
fix pearl create multiple threads

### PR DESCRIPTION
**Pull request by idruzhitskiy**

### CODE REVIEW

File: src/core/backend/pearl/group.rs [hunk 1]
```diff
@@ -1,16 +1,17 @@
 use super::prelude::*;
 use ring::digest::{digest, SHA256};
+use std::collections::HashMap;
```
Comment (agend) [inline src/core/backend/pearl/group.rs:3]: Does it worth to move to prelude ?

File: src/core/backend/pearl/group.rs [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
 
 #[derive(Clone, Debug)]
 pub(crate) struct Group {
     holders: Arc<RwLock<Vec<Holder>>>,
-    pearl_sync: Arc<SyncState>,
```
Comment (agend) [inline src/core/backend/pearl/group.rs:7]: Show we delete `SyncState` trait then?

File: src/core/backend/pearl/group.rs [hunk 3]
```diff
@@ -121,32 +123,40 @@ impl Group {
...
-        self.settings
-            .config()
-            .try_multiple_times_async(
-                || self.try_create_write_pearl(timestamp),
-                "pearl init failed",
-                self.settings.config().settings().create_pearl_wait_delay(),
-            )
+    async fn create_write_pearl(&self, ts: u64) -> BackendResult<Holder> {
+        if !self.created_holder_indexes.read().await.contains_key(&ts) {
+            let mut indexes = self.created_holder_indexes.write().await;
```
Comment (agend) [inline src/core/backend/pearl/group.rs:128]: Could it be here a Deadlock if you take read lock right before?

File: src/core/backend/pearl/group.rs [hunk 4]
```diff
@@ -121,32 +123,40 @@ impl Group {
...
-    async fn create_write_pearl(&self, timestamp: u64) -> BackendResult<Holder> {
-        self.settings
-            .config()
-            .try_multiple_times_async(
-                || self.try_create_write_pearl(timestamp),
-                "pearl init failed",
-                self.settings.config().settings().create_pearl_wait_delay(),
-            )
+    async fn create_write_pearl(&self, ts: u64) -> BackendResult<Holder> {
+        if !self.created_holder_indexes.read().await.contains_key(&ts) {
```
Comment (agend) [inline src/core/backend/pearl/group.rs:127]: aren't here a race condition?

File: src/core/backend/pearl/group.rs (hunk 3 above)
Comment (idruzhitskiy) [reply src/core/backend/pearl/group.rs:128]: Readlock gets dropped, isn't it?

Comment (agend) [reply src/core/backend/pearl/group.rs:126]: @pyakushin what do you think.
@idruzhitskiy what about race condtion?

**Pull request merged**